app/src/main/java/org/mozilla/fenix/settings/advanced/LocaleViewHolders.kt

```diff
@@ -20,10 +20,22 @@ class LocaleViewHolder(
 ) : BaseLocaleViewHolder(view, selectedLocale) {
 
     override fun bind(locale: Locale) {
-        // Capitalisation is done using the rules of the appropriate locale (endonym and exonym).
-        locale_title_text.text = locale.getDisplayName(locale).capitalize(locale)
-        // Show the given locale using the device locale for the subtitle.
-        locale_subtitle_text.text = locale.displayName.capitalize(Locale.getDefault())
+        if (locale.language == "zh" && locale.country == "CN") {
+            locale_title_text.text =
+                Locale.forLanguageTag("zh-Hans").getDisplayName(locale).capitalize(locale)
+            locale_subtitle_text.text =
+                Locale.forLanguageTag("zh-Hans").displayName.capitalize(Locale.getDefault())
+        } else if (locale.language == "zh" && locale.country == "TW") {
+            locale_title_text.text =
+                Locale.forLanguageTag("zh-Hant").getDisplayName(locale).capitalize(locale)
+            locale_subtitle_text.text =
+                Locale.forLanguageTag("zh-Hant").displayName.capitalize(Locale.getDefault())
```

Comment:
Can we break this part out into a helper function so that it is easier to read? 

Suggestion: 

```kotlin
override fun bind(locale: Locale) {
     if (locale.language == "zh") {
            handleChineseLocale()
     } else {
            // Capitalisation is done using the rules of the appropriate locale (endonym and exonym).
            locale_title_text.text = locale.getDisplayName(locale).capitalize(locale)
            // Show the given locale using the device locale for the subtitle.
            locale_subtitle_text.text = locale.displayName.capitalize(Locale.getDefault())
     } 
}

... 

private fun handleChineseLocale() {
        if (locale.country == "CN") {
            locale_title_text.text =
                Locale.forLanguageTag("zh-Hans").getDisplayName(locale).capitalize(locale)
            locale_subtitle_text.text =
                Locale.forLanguageTag("zh-Hans").displayName.capitalize(Locale.getDefault())
        } else if (locale.country == "TW") {
            locale_title_text.text =
                Locale.forLanguageTag("zh-Hant").getDisplayName(locale).capitalize(locale)
            locale_subtitle_text.text =
                Locale.forLanguageTag("zh-Hant").displayName.capitalize(Locale.getDefault())
}
```